app/main/posts/views/filters/filters-dropdown.directive.js

```diff
@@ -29,12 +29,9 @@ function FiltersDropdown(PostFilters, ModalService, $rootScope, _, $location, Sa
         };
         // Check if we can edit
         function setSavedSearchUpdateStatus() {
-            var savedSearchId = PostFilters.getModeId();
-            if (savedSearchId) {
-                SavedSearchEndpoint.get({id: savedSearchId}, function (savedSearch) {
-                    $scope.canUpdateSavedSearch = _.contains(savedSearch.allowed_privileges, 'update');
-                });
-            }
+            var savedSearch = PostFilters.getModeEntity();
+
+            $scope.canUpdateSavedSearch = savedSearch && _.contains(savedSearch.allowed_privileges, 'update');
```

Comment:
IDK if this is worth it. Maybe we would prefer to go and get it , but since it's already available , not sure